implementing_seraphis/content.tex

```diff
@@ -606,38 +658,46 @@ \subsection{Jamtis wallets}
+    \item \textbf{Payment validator} ($d_{vr}, K_s$): In addition to the find-received and generate-address tier capabilities, the payment validator can perform balance recovery for normal enotes up to (but not including) the key image computation.
 
     \item \textbf{View-balance} ($k_{vb}, k_m U$): View all balance-related information including for selfsend enotes.
 
     \item \textbf{Master} ($k_{vb}, k_m$): View the entire user's balance and make ownership/unspentness proofs for all enotes.
 \end{itemize}
 
-Here we see the value of $k_{ua}$, which separates payment validators from the find-received and generate-address wallet tiers.
+Here we see the value of $d_{vr}$, which separates payment validators from the filter-assist and generate-address wallet tiers.
 
 \subsubsection{Remote-assisted balance recovery}
 \label{subsubsec:jamtis-wallets-remote-assisted}
 
-A user can set up a remote find-received wallet that creates basic records from enotes in the ledger and passes those records to their local view-balance wallet (or payment validator). There are several important details to highlight.
+A user can set up a remote filter-assist wallet that passes filtered enotes to their local view-balance wallet (or payment validator). There are several important details to highlight.
 \begin{itemize}
-    \item A find-received wallet has very limited information about the user's owned enotes.
-    \begin{enumerate}
-        \item It can perform view tag checks on all enotes, which essentially filters down the set of enotes `possibly owned by the user'.
-        \item It can uncover the nominal address tags of normal enotes, which means if the wallet knows about (or ever in the future learns about) addresses owned by the user, then it can identify all normal enotes owned by those addresses.
-    \end{enumerate}
+    \item A filter-assist wallet has very limited information about the user's owned enotes. It can only perform primary tag checks on enotes, which merely filters down the set of enotes `possibly owned by the user'.
 
-    \item A find-received wallet only has to transmit basic records of enotes that pass view tag checks and key images associated with those enotes to the user. This amounts to a substantial relative reduction in data that a user needs to access compared to the remote wallet, even if view tags are only 8 bits (a 256:1 reduction).
+    \item A filter-assist wallet only has to transmit enotes that pass primary view tag checks and all other enotes in those enotes' transactions (since they may pass the auxiliary view tag check, which requires $k_{vb}$), and key images associated with those enotes to the user. This amounts to a substantial relative reduction in data that a user needs to access compared to the remote wallet, even if primary view tags are only 8 bits (a ~128:1 reduction).
```

Comment:
Yes, this is part of the auxiliary enotes design. This part is orthogonal to hiding the nominal address tag and flexible view tags, but would be an enhancement that lets light wallets do churning and pocket change with much greater privacy. The strain on full wallets is minimal, since full wallets already have to download all enotes and try doing an X25519 exchange on each. Full wallets would have to perform an additional SHA3 hash on enotes in the same transaction as an enote which passes the primary view tag check, but that would only slow down their computation by about 1%. However, light wallets would have to download and try hashing for an auxiliary self send type for about 2x as many enotes, at the benefit of being able to split and churn their funds with no additional loss of privacy as compared to a normal transaction that they are involved in. For more technical context about this issue, I recommend reading [this comment](https://gist.github.com/tevador/50160d160d24cfc6c52ae02eb3d17024?permalink_comment_id=4706509#gistcomment-4706509).

Reply:
I will admit that I didn't do a very good job at explaining in the paper....